runners/samza/src/main/java/org/apache/beam/runners/samza/translation/ParDoBoundMultiTranslator.java

```diff
@@ -301,12 +304,18 @@ public void translatePortable(
 
     final RunnerApi.PCollection input = pipeline.getComponents().getPcollectionsOrThrow(inputId);
     final PCollection.IsBounded isBounded = SamzaPipelineTranslatorUtils.isBounded(input);
+    final Coder<?> keyCoder =
+        StateUtils.isStateful(stagePayload)
```

Comment:
same above. consider adding a getKeyCoder() in stateutils.

Reply:
This logic to extract key coder from a portable pipeline is only used here. It is different from class pipeline approach and depends on different inputs.